doc/demo/sparse_train/sparse_train.md

```diff
@@ -0,0 +1,291 @@
+# Sparse Training with PaddlePaddle
+
+This Document will guide you to understand the sparse design in PaddlePaddle.
+
+## Quick Start
+
+Assuming that you are familiar with PaddlePaddle. Here show train billion level parameters with sparse training. 
+
+### Prepare Data
+Given the Input data sample as follows:
+```
+0,0,7216737 1,18127654 1, ... 
+```
+The first slot ```0``` is just one flag, not one feature; the second slot ```0``` is  label; the third slot is feature id, ```7216737 1```means the ```7216737``` dimension is not zero, the ```1``` is just one flag, not one feature; the ```18127654 1``` is another non zero feature. The line data depicts one simple high dimension input data for classification task.
+
+All samples are stored as simple TXT file ```train.txt```, then set it in ```train.list``` index file, which is used in model configuration file. We just provides no more than 2000+ samples to explains following  experiments. Check related HOWTO in other DOC. 
+
+### Prepare DataProvider
+Use following dataprovider to feed DataProvider:
+```python
+# Define a py data provider
+@provider(input_types=[
+    sparse_binary_vector(18182296),
+    integer_value(2)
+])
+def process(settings, filename):
+    f = open(filename, 'r')
+    for line in f:  # read each line
+        splits = line.split(',')
+        label = int(splits[1])
+        splits.pop(0)
+        splits.pop(0)
+        sparse_non_values = []
+        for value in splits:
+            v = value.split(" ")
+            sparse_non_values.append(long(v[0]))
+        # give data to paddle.
+        yield sparse_non_values, label
+    f.close()  # close file
+```
+Please consult related DOC for ```sparse_binary_vector``` and ```integer_value```.
+
+### Prepare Model Configuration
+Here providing one simple fc_layers network, which is enough for explaining HOW to train DNN with sparse training.
+
+We use ```ParameterAttribute(sparse_update=True)``` to enable sparse training for local job as well as cluster job. The ```trainer_config_helpers``` model will set internal FLAGs automatically. At last, these layers not set with sparse FLAG do dense training while sparse layers automatically use sparse training. Generally RECOMMEND you set these layers with high sparsity input, such as first hidden layers with sparse input.
```

Comment:
This detail can help user understand what happened in internally, which is prefered in this DOC. We hope users understand that the Sparse Training is not only several sparse Matrix muliplication, but also involving forwardbackward,optimisation, some additional design for cluseter.